### ISSUE TITLE
Cannot save document after sharer has added edits

### ISSUE THREAD

**Issue by lgnch** [nextcloud/richdocuments#780]
### Steps to reproduce
1. Share a document or older
2. Have user create a new document or simply edit an existing one
3. User will not be able to save if editing an existing file
4. If editing a new file user can save ok., But the folder sharer edits the file and saves then the shared user can no longer save.

### Expected behaviour
user who is shared a file should be able to edit and save the file while it is shared with edit permissions.

### Actual behaviour
user cannot save the file. Document errors with "document cannot be saved. Check your permissions or contact the storage server adminstrator."

This error has only begin since the update to collbora
Package loolwsd-4.2.0-3.el7.x86_64
Package CODE-brand-4.2-12.noarch

### Server configuration

**Operating system**:
Centos 7

**Web server:**
Apache 2.4.6

**Database:**
Mariadb 

**PHP version:**
PHP 7.2.26

**Nextcloud version:** (see Nextcloud admin page)
Nextcloud 17.0.2

**Updated from an older Nextcloud/ownCloud or fresh install:**
Upgrade and fresh cause same issue

**Where did you install Nextcloud from:**
Git Hub

**Signing status:**
<details>
<summary>Signing status</summary>

```
Login as admin user into your Nextcloud and access 
http://example.com/index.php/settings/integrity/failed 
paste the results here.
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your Nextcloud installation folder

or 

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```
</details>

**Are you using external storage, if yes which one:** local/smb/sftp/...

**Are you using encryption:** yes/no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...

#### LDAP configuration (delete this part if not used)
<details>
<summary>LDAP config</summary>

```
With access to your command line run e.g.:
sudo -u www-data php occ ldap:show-config
from within your Nextcloud installation folder

Without access to your command line download the data/owncloud.db to your local
computer or access your SQL server remotely and run the select query:
SELECT * FROM `oc_appconfig` WHERE `appid` = 'user_ldap';

Eventually replace sensitive data as the name/IP-address of your LDAP server or groups.
```
</details>

### Client configuration
**Browser:**

**Operating system:**

### Logs
#### Web server error log
<details>
<summary>Web server error log</summary>
```
Insert your webserver log here
```
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
Insert your Nextcloud log here
```
</details>

#### Browser log
<details>
<summary>Browser log</summary>
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
</details>

**Comment by lgnch** [comment]
Error in logs shows an Exception: Call to a member function getId() on boolean

[index] Error: Exception: Call to a member function getId() on boolean at <<closure>>

0. /var/www/nextcloud/lib/private/AppFramework/App.php line 126
   OC\AppFramework\Http\Dispatcher->dispatch(OCA\Richdocument ... {}, "putFile")
1. /var/www/onthemove/lib/private/AppFramework/Routing/RouteActionHandler.php line 47
   OC\AppFramework\App::main("OCA\\Richdocume ... r", "putFile", OC\AppFramework\ ... {}, {fileId: "161925 ... "})
2. <<closure>>
   OC\AppFramework\Routing\RouteActionHandler->__invoke({fileId: "161925 ... "})
3. /var/www/nextcloud/lib/private/Route/Router.php line 297
   undefinedundefinedcall_user_func(OC\AppFramework\ ... {}, {fileId: "161925 ... "})
4. /var/www/nextcloud/lib/base.php line 997
   OC\Route\Router->match("/apps/richdocum ... s")
5. /var/www/nextcloud/index.php line 42
   OC::handleRequest()

POST /index.php/apps/richdocuments/wopi/files/1619259_ockmln9o7u3e/contents?access_token=REMOVED&access_token_ttl=0

This maybe the same issue already tracked, or similar too, in 753. I am reporting as it has just started after the recent update from Collabora which we patched today.

**Comment by lgnch** [comment]
Same as issue 781
App version: Collabora Online 3.5.1

**Comment by lgnch** [comment]
Confirmed, a return to 3.5.0 solves the issue.

**Comment by juliushaertl** [comment]
Could you give some more details about the files? Are they stored on external storage? Shared with another user?

**Comment by lgnch** [comment]
Files are stored locally as well as on external storage (NAS - Synology RS816 with proper write permissions, has been running with no issues until this update for 3 years), error occurs in both situations. Browsers tested are Chrome and Firefox. Latest version of CODE - LooLWSD 4.2.0 installed with collabora office 6.2-3.

**Comment by lgnch** [comment]
To answer the last question, yes, files are shared to users. Error also occurs when shared to groups.

**Comment by juliushaertl** [comment]
Any chance you could give the patch at https://github.com/nextcloud/richdocuments/pull/805 a try and see if that fixes the issue for you?

**Comment by lgnch** [comment]
I will try today and come back to you with an answer.

**Comment by lgnch** [comment]
This did not resolve it. In fact,m it created a new problem where, when a file is shared for access and parties edit it, the file once edited and saved, errors and saves it as a new document. See the attached

![errro1](https://user-images.githubusercontent.com/4876162/73302181-0de7ea00-4279-11ea-917b-96640a8f210d.png)

The document then has to be reloaded as a revision which does not have the changes in it.
 image. If editors are working together, the sharer can save, but the new person, if the sharer leaves the document, is presented with the attached and has to save it as a new document that they then need to share out.

So short, answer, NO.

**Comment by juliushaertl** [comment]
Thanks a lot for giving this a test run, this is highly appreciated. That was indeed an issue I didn't catch. Could you give the two additional commits in https://github.com/nextcloud/richdocuments/pull/805 another try.

**Comment by lgnch** [comment]
Will execute tests later today.

**Comment by lgnch** [comment]
Changes tested. This has resolved the saving issues in #780. tests involved individual edits and saves to same file with users editing and exiting the document. Then saves where tested on the fly with users editing at the same time. Both worked without error on this commit.

**Comment by lgnch** [comment]
New issue has appeared and is reported in #806

**Comment by juliushaertl** [comment]
Fixed with #805

**Comment by juliushaertl** [comment]
This should be fixed with the new 3.5.2 release.
